### PR TITLE
Make GenerateType work off of an Incomplete-Member.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-compiler @dotnet/roslyn-ide

**Comment by CyrusNajmabadi** [comment]
Fixes https://github.com/dotnet/roslyn/issues/13211

Note: this is still an issue because of https://github.com/dotnet/roslyn/issues/7536   @gafter   Do we think we can get around to fixing the issue where the compiler does not bind incomplete members?

**Comment by CyrusNajmabadi** [comment]
@dotnet/roslyn-compiler Please review simple change to SyntaxFacts.  Thanks!

**Comment by gafter** [comment]
Compiler changes LGTM.

I'll try to fix  #7536 sooner rather than later, but C# 7 changes have to take priority.

File: src/Compilers/CSharp/Portable/Syntax/SyntaxFacts.cs [hunk 1]
```diff
@@ -195,6 +195,9 @@ public static bool IsInTypeOnlyContext(ExpressionSyntax node)
 
                     case TypedVariableComponent:
                         return ((TypedVariableComponentSyntax)parent).Type == node;
+
+                    case IncompleteMember:
+                        return ((IncompleteMemberSyntax)parent).Type == node;
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/Syntax/SyntaxFacts.cs]: > return ((IncompleteMemberSyntax)parent).Type == node; [](start = 23, length = 54)

Does this change the way SemanticModel APIs behave? Please add compiler tests for them and a direct test for IsInTypeOnlyContext API.  src\Compilers\CSharp\Test\Symbol\Compilation\SemanticModelGetSemanticInfoTests.cs looks like a good place.

Comment (CyrusNajmabadi) [inline src/Compilers/CSharp/Portable/Syntax/SyntaxFacts.cs]: Will do, thanks @AlekseyTs .  If i make those changes, are you ok with this PR?

Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/Syntax/SyntaxFacts.cs]: Yes, other than that, LGTM.

**Pull request merged**